docs/user-guide/cabal-projects.md

```diff
@@ -81,7 +81,7 @@ instantiate a package set.
 # default.nix
 let
   # Import the Haskell.nix library,
-  haskell = import (builtins.fetchTarball https://github.com/input-output-hk/haskell.nix/archive/master.tar.gz) {};
+  haskell = import (builtins.fetchTarball https://github.com/input-output-hk/haskell.nix/archive/master.tar.gz);
```

Comment:
```suggestion
  haskell =
    (let src = builtins.fetchTarball https://github.com/input-output-hk/haskell.nix/archive/master.tar.gz;
      in import (src + "/nixpkgs") (import src)).haskell-nix;
```
after trying this in the `nix-repl`, should this really be like this?

Reply:
You're probably right. I don't really remember how haskell.nix was imported before the change to an overlay. I assumed it was passed to `import nixpkgs` back then too, since that was missing here I left it out.
Should I apply this patch throughout?

Reply:
Yes please :-)

Reply:
I applied a minor variation of your suggestion. I do use the imported `nixpkgs` in my own projects using `haskell.nix` and I figured people might copy the example verbatim and this makes that a nicer experience imo. Let me know if you prefer the version that brings fewer names into scope. Maybe `src` should be `haskell-nix-src` or be local to the `nixpkgs` definition to avoid consuming a popular name?

I rebased while I was at it, let me know if this interferes with reviewing and I should avoid it in the future.